pypodio2/transport.py

```diff
@@ -48,6 +48,24 @@ def __call__(self):
         return self.token.to_headers()
 
 
+class OAuthRefreshTokenAuthorization(object):
+    """Generates headers for Podio OAuth2 Authorization from a refresh token."""
+
+    def __init__(self, client_id, client_secret, refresh_token, domain):
+        body = {'grant_type': 'refresh_token',
+                'client_id': client_id,
+                'client_secret': client_secret,
+                'refresh_token': refresh_token}
+        http = Http(disable_ssl_certificate_validation=True)
+        headers = {'content-type': 'application/x-www-form-urlencoded'}
+        response, data = http.request(domain + '/oauth/token', 'POST',
+                                      urlencode(body), headers=headers)
+        self.token = OAuthToken(_handle_response(response, data))
+
+    def __call__(self):
+        return self.token.to_headers()
```

Comment:
Now I'm puzzled. The name of `OAuthRefreshTokenClient` makes it look like a class, but it's a function. OTOH `OAuthRefreshTokenAuthorization` is a callable class. What's the rationale here?

Reply:
Same case here, I'm just following the existing code style 😞 I don't think it's worth to refactor all the original code. But you are correct, the original code is a bit puzzling at parts.